projects/packages/my-jetpack/src/products/class-product.php

```diff
@@ -0,0 +1,185 @@
+<?php
+/**
+ * Base product
+ *
+ * @package my-jetpack
+ */
+
+namespace Automattic\Jetpack\My_Jetpack;
+
+use Automattic\Jetpack\Plugins_Installer;
+
+/**
+ * Class responsible for handling the products
+ */
+abstract class Product {
+
+	/**
+	 * The product slug
+	 *
+	 * @var string
+	 */
+	public static $slug = null;
+
+	/**
+	 * The filename (id) of the plugin associated with this product. If not defined, it will default to the Jetpack plugin
+	 *
+	 * @var string
+	 */
+	public static $plugin_filename = null;
+
+	/**
+	 * The slug of the plugin associated with this product. If not defined, it will default to the Jetpack plugin
+	 *
+	 * @var string
+	 */
+	public static $plugin_slug = null;
+
+	/**
+	 * The Jetpack plugin slug
+	 *
+	 * @var string
+	 */
+	const JETPACK_PLUGIN_SLUG = 'jetpack';
+
+	/**
+	 * The Jetpack plugin filename
+	 *
+	 * @var string
+	 */
+	const JETPACK_PLUGIN_FILENAME = 'jetpack/jetpack.php';
+
+	/**
+	 * Get the Product info for the API
+	 *
+	 * @throws \Exception If required attribute is not declared in the child class.
+	 * @return array
+	 */
+	public static function get_info() {
+		if ( is_null( static::$slug ) ) {
+			throw new \Exception( 'Product classes must declare the $slug attribute.' );
+		}
+		return array(
+			'slug'        => static::$slug,
+			'name'        => static::get_name(),
+			'description' => static::get_description(),
+			'status'      => static::get_status(),
+			'class'       => get_called_class(),
```

Comment:
usually we use `get_info()` to expose data. Do you think we should expose the class name there, too? I see this data is propagated in the client, maybe something that we wouldn't like to do.

<img src="https://user-images.githubusercontent.com/77539/151792131-7686c4ca-33b1-43f9-aac3-c35c4a0126a4.png" width="400px" />

Reply:
oh ic, the class is defined by the client :-/. So when the request doesn't contain the handler class, it will return `The product class handler is not implemented` error.

Reply:
Maybe we can map the handler according to the product slug?

Reply:
we can remove this from the json, yes, No need for it to be there